src/Content/Version.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+namespace Kirby\Content;
+
+use Kirby\Cms\ModelWithContent;
+use Kirby\Exception\NotFoundException;
+
+/**
+ * The Version class handles all actions for a single
+ * version and is identified by a VersionId instance
+ *
+ * @internal
+ * @since 5.0.0
+ *
+ * @package   Kirby Content
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ */
+class Version
+{
+	public function __construct(
+		protected ModelWithContent $model,
+		protected VersionId $id
+	) {
+	}
+
+	/**
+	 * Returns a Content object for the given language
+	 */
+	public function content(string $language = 'default'): Content
+	{
+		return new Content(
+			parent: $this->model,
+			data:   $this->model->storage()->read($this->id, $language),
+		);
+	}
+
+	/**
+	 * Creates a new version for the given language
+	 */
+	public function create(array $fields, string $language = 'default'): void
+	{
+		$this->model->storage()->create($this->id, $language, $fields);
+	}
+
+	/**
+	 * Deletes a version by language or for any language
+	 */
+	public function delete(string|null $language = null): void
+	{
+		// delete all languages
+		if ($language === null) {
+			foreach ($this->model->kirby()->languages() as $language) {
+				$this->model->storage()->delete($this->id, $language->code());
+			}
+		}
+
+		// delete the default language in single-language mode
+		if ($this->model->kirby()->multilang() === false) {
+			$this->model->storage()->delete($this->id, 'default');
+			return;
+		}
+
+		// delete a single language
+		$this->model->storage()->delete($this->id, $language);
```

Comment:
What is the use case for deleting a single language? Do we already support that in v4?

Reply:
Tbh I mostly just moved over what we had in ContentStorage here. I think you are right and we don't need it so far. This would also clean up the method some more. I'm not super happy about the ambiguous usage depending on what's passed.

Reply:
I'll check the code and concept history again. If there is no reason, I'll remove it. We can still add it back if we ever need it.

Reply:
Did you have time to check it already or should I have another look at it?

Reply:
Checked the concept and code again. What I could find out:

- In the concept, we only had the handlers, not `ContentStorage`. There we only intended to support deleting a specific language. A `null` param to the `$handler->delete()` method for looping over all languages was not supported.
- Reasoning as far as I remember was again to prepare for a future where languages are independent. So you may then want to delete a single language.
- The `ContentStorage` class introduced the nullable `$lang` param. But if I understand the `ContentStorage` logic correctly, it does not mean "all languages" but instead "default language". The `null` is passed to the `$contentStorage->language()` method and that passes it to `$app->language(null)` in multi-lang or just returns `default` in single-lang.

**TL;DR:** If I understand it correctly, the nullable param as you have implemented it here is a new concept we didn't have in the current code base.

`Suggestion`: As the `Version` class is internal anyway, we only need to support what we need in the core. And so far that's only "all languages". "Specific language" starts getting relevant with the new translation concept only.

Reply:
I think your suggestion makes a lot of sense. Can we also handle it in a way that we combine it with a later refactoring after all the other PRs and add a todo for it to avoid a lot of rebases?

Reply:
Added it to the todo list